config/src/network_id.rs

```diff
@@ -143,6 +139,17 @@ impl NetworkId {
     pub fn vfn_network() -> NetworkId {
         NetworkId::Private("vfn".to_string())
     }
+
+    /// Return the NetworkId as a copy-on-write str slice. Certain variants are
+    /// constant, so we can just return a `&'static str` instead of allocating
+    /// an extra temporary string.
+    pub fn as_cow_str(&self) -> Cow<'static, str> {
+        match self {
+            NetworkId::Validator => Cow::Borrowed("Validator"),
+            NetworkId::Public => Cow::Borrowed("Public"),
+            NetworkId::Private(info) => Cow::Owned(format!("Private({})", info)),
```

Comment:
Realistically, we could probably just print the `info` and ignore the Private part

Reply:
agreed, we could easily make this just print the inner `info`. would also make this more efficient

i guess someone could name their private network id "Validator", but that would be their fault haha

Reply:
Same as above. We may not make a copy here, but we make a ton more copies to route this to the eventual logs.